sdk/core/Azure.Core/src/SequenceWriter.cs

```diff
@@ -0,0 +1,166 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Buffers;
+using System.IO;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Azure.Core
```

Comment:
We shoudl add a warning to this type describing the dangers of misusing this type after Dispose is called.